docs/APIClient.md

```diff
@@ -0,0 +1,208 @@
+if information is missing (and subsequently storing the results of those questions in the RC file for future lookups).
+
+An example use of the config client:
+
+```
+from timesketch_api_client import config
+
+ts_client = config.get_client()
+```
+
+To be able to take advantage of the config client, the user needs to be running this from the command line or in a way where questions can be asked
+and answered (or the config file to be fully populated). This works both in CLI scripts as well as in a notebook.
+
+By default the client credentials will be stored on disk in a Token file, in an encrypted format, protected by a randomly generated
+password that is stored in the RC file. It is highly recommended to protect your credential file using a strong password. The password
+for the file can be passed on to the config class using:
+
+```
+ts_client = config.get_client(token_password='MY_SUPER_L337_PWD')
+```
+
+If the token file does not exist, it will be generated and encrypted using the supplied password.
+
+### Using the Timesketch Client
+
+The TS client only has limited functionality, such as to list open indices, what sketches the user has access to as well as to fetch sketches, indices, etc.
+The client object also contains functions to check for OAUTH token status and refresh it, if needed.
+
+Most of the functionality of the API client lies in sketch operations. To list all available sketches the function `list_sketches` can be used:
+
+```
+for sketch in ts_client.list_sketches():
+  # Do something with the sketches...
+```
+
+To print the name and description of all available sketches, something like:
+
+```
+for sketch in ts_client.list_sketches():
+  print('[{0:d}] {1:s} <{2:s}>'.format(sketch.id, sketch.name, sketch.description))
+```
+
+## Connecting to a Sketch
+
+There are two ways of getting a sketch object, either by listing all available sketches or by fetching a specific one.
+To fetch a specific one use the `get_sketch` function:
+
+```
+sketch = ts_client.get_sketch(SKETCH_ID)
+```
+
+All you need is the sketch ID, eg:
+
+```
+sketch = ts_client.get_sketch(25)
+```
+
+## Overview of a Sketch
+
+A sketch object has few properties that can be used to get a better overview of what it contains:
+
++ **acl**: A python dict with the current sketch ACL.
++ **labels**: A list of strings, with all the labels that are attached to the sketch.
++ **name**: The name of the sketch.
++ **description**: The description of the sketch.
++ **status**: The status of the sketch.
+
+Each of these properties can be accessed using `sketch.<PROPERTY>`, eg. `sketch.name`, or `sketch.labels`
+
+## Explore Data
+
+The sketch object has several ways to explore data, via aggregations or searches.
+
+### Views
+
+The first is simply to list up all views... the function `list_views` returns back View object for all views. An example overview would be:
```

Comment:
"To list all the views use the `list_views` function. This functions returns a View object for all views....."